includes/class-wc-payments-order-service.php

```diff
@@ -794,4 +795,52 @@ private function update_order_status( $order, $order_status, $intent_id = '' ) {
 			Logger::log( 'Error when updating status for order ' . $order->get_id() . ': ' . $e->getMessage() );
 		}
 	}
+
+	/**
+	 * Return terminal intent payment method array based on payment methods request.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @param array           $default_value - default value.
+	 *
+	 * @return array|null
+	 * @throws \Exception
+	 */
+	public function get_terminal_intent_payment_method( $request, array $default_value = [ Payment_Method::CARD_PRESENT ] ) :array {
+		if ( null === $request->get_param( 'payment_methods' ) ) {
+			return $default_value;
+		}
+
+		if ( ! is_array( $request->get_param( 'payment_methods' ) ) ) {
+			throw new \Exception( 'Invalid param \'payment_methods\'!' );
+		}
+
+		foreach ( $request->get_param( 'payment_methods' ) as $value ) {
+			if ( ! in_array( $value, Payment_Method::ALLOWED_PAYMENT_METHODS, true ) ) {
+				throw new \Exception( 'Invalid param \'payment_methods\'!' );
```

Comment:
Should we instead return a more specific message like "one or more payment methods are not supported."